src/ucp/core/ucp_rkey.c

```diff
@@ -52,14 +61,28 @@ ucs_status_t ucp_rkey_pack(ucp_context_h context, ucp_mem_h memh,
         ++uct_memh_index;
         p += pd_size;
     }
+    if (uct_memh_index == 0) {
+        status = UCS_ERR_UNREACHABLE;
```

Comment:
Would not UCS_ERR_UNSUPPORTED be more relevant here? Or maybe add a new one then, like UCS_ERR_MEM_REGISTRATION..